systemd/cgroups.go

```diff
@@ -0,0 +1,258 @@
+package systemd
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/log"
+	"golang.org/x/sys/unix"
+)
+
+// CgroupUnified constant values describe how cgroup filesystems (aka hierarchies) are
+// mounted underneath /sys/fs/cgroup. In cgroups-v1 there are many mounts,
+// one per controller (cpu, blkio, etc) and one for systemd itself. In
+// cgroups-v2 there is only one mount managed entirely by systemd and
+// internally exposing all controller syscalls. As kernel+distros migrate towards
+// cgroups-v2, systemd has a hybrid mode where it mounts v2 and uses
+// that for process management but also mounts all the v1 filesystem
+// hierarchies and uses them for resource accounting and control
+type CgroupUnified int8
+
+const (
+	// CgroupUnifiedNone indicates that both systemd and the controllers
+	// are using v1 legacy mounts and there is no usage of the v2
+	// unified hierarchy. a.k.a "legacy hierarchy"
+	CgroupUnifiedNone CgroupUnified = iota
+	// CgroupUnifiedSystemd indicates that systemd is using a v2 unified
+	// hierarcy for organizing processes into control groups, but all
+	// controller interaction is using v1 per-controller hierarchies.
+	// a.k.a. "hybrid hierarchy"
+	CgroupUnifiedSystemd CgroupUnified = iota
+	// CgroupUnifiedAll indicates that v2 API is in full usage and there
+	// are no v1 hierarchies exported. Programs (mainly container orchestrators
+	// such as docker,runc,etc) that rely on v1 APIs will be broken.
+	// a.k.a. "unified hierarchy"
+	CgroupUnifiedAll CgroupUnified = iota
+)
+
+// WARNING: We only read this data once at process start, systemd updates
+// may require restarting systemd-exporter
+var cgroupUnified *CgroupUnified = nil
+
+// Values copied from https://github.com/torvalds/linux/blob/master/include/uapi/linux/magic.h
+const (
+	TMPFS_MAGIC         = 0x01021994
```

Comment:
Consider naming it "go way" i.e. TmpfsMagic as per https://talks.golang.org/2014/names.slide#5, https://talks.golang.org/2014/names.slide#5

Reply:
I think we should leave this one as-is, given that the name was borrowed directly from a linux header library. If I was copying all of the values (e.g. creating a go package for using magic fs numbers) then I would agree, but in this case I thought it was useful to keep the reference format. 

PS - I considered trying to import this value, but could not find a library (and we only need these three values anyway)